### PR TITLE
add refresh button

### PR DESCRIPTION
**Pull request by ChiChiHeHeZhou**
## What this PR does / why we need it

Add refresh button for Vsphere
1.Refresh button will call vsphere API to update os images list
image

2.Show loading icon while wait for the results
image

## Which issue(s) this PR fixes

Fixes: #5184

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx [hunk 1]
```diff
@@ -188,26 +220,44 @@ function VmTemplateSection(
...
-                <select {...register(VSPHERE_FIELDS.VMTEMPLATE)} onChange={handleOsImageSelect}>
+        <CdsFormGroup layout="vertical-inline" control-width="shrink">
+            <div cds-layout="horizontal gap:lg align:vertical-center">
+                <SpinnerSelect
+                    className="select-md-width"
+                    disabled={false}
+                    label="OS Image"
+                    handleSelect={handleOsImageSelect}
+                    name={VSPHERE_FIELDS.VMTEMPLATE}
+                    controlMessage="OS Images will be retrieved after selecting Datacenters"
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx:241]: nitpick: please change to "OS Images will be retrieved after selecting a Datacenter"

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx [hunk 2]
```diff
@@ -130,6 +136,24 @@ export function VsphereClusterSettingsStep(props: Partial<StepProps>) {
...
+    };
+
+    const fetchOsImages = async () => {
+        try {
+            setOsImagesLoading(true);
+            await VsphereService.getVSphereOsImages(vsphereState[STORE_SECTION_FORM][VSPHERE_FIELDS.DATACENTER]).then((fetchedOsImages) => {
+                setOsImages(fetchedOsImages);
+            });
+        } catch (e: any) {
+            console.log(`Unabled to get ec2 key pair: ${e}`);
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx:151]: error log should say "Unable to retrieve OS Images"

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx (hunk 1 above)
Comment (ChiChiHeHeZhou) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx:241]: Will do

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx (hunk 2 above)
Comment (ChiChiHeHeZhou) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereClusterSettingsStep.tsx:151]: Right sir.

**Review by miclettej — APPROVED** [review]

**Pull request merged**